components/bl-modal-component/src/index.js

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function ModalComponent({ component, eventHandlers, pods }) {
+  const { display, classList, style, disabled, modalVisibility } = component;
+  const { onClose } = eventHandlers;
+
+  const modalContentPod = pods['modalContent'];
+
+  useEffect(() => {
+    const handleEscClick = e => {
+      if (e.keyCode === 27) {
+        if (onClose) {
+          onClose({ modalVisibility: false });
+        }
+      }
+    };
+
+    document.addEventListener('keydown', handleEscClick);
+
+    return () => document.removeEventListener('keydown', handleEscClick);
+  }, []);
```

Comment:
how about create a custom hook useCloseOnEscape and move that logic to the bottom of the component? That could make it more readable.